src/io/iohk/decco/test/snappy/SnappySpec.scala

```diff
@@ -0,0 +1,25 @@
+package io.iohk.decco.snappy
+
+import io.iohk.decco.test.utils.CodecTestingHelpers._
+import io.iohk.decco.definitions._
+import io.iohk.decco.snappy.auto._
+import org.scalatest.{FlatSpec, Matchers}
+import io.iohk.decco.BufferInstantiator.global.HeapByteBuffer
+
+class SnappySpec extends FlatSpec with Matchers {
```

Comment:
Just one question. Presumably this is now the snappy codec compressing the result of the decco.auto encoding scheme? How is this working? How would I make snappy work with a different encoding scheme?

Reply:
There are two layers: first is codec intrinsics defined in a class implementing `CodecContract`, another is codec outer interface exposed as `Codec`. If you want to provide your own encoding scheme, you should implement implicit `CodecContract` and Snappy will use it. Just don't import `io.iohk.decco.auto` which has implicit conversion `CodecContract => Codec`.